server/webpa.go

```diff
@@ -257,6 +274,37 @@ func (w *WebPA) Prepare(logger log.Logger, health *health.Health, primaryHandler
 			ListenAndServe(logger, &w.Alternate, alternateServer)
 		}
 
+		if metricsServer := w.Metrics.New(logger, getMetricsHandler(w.Gatherer, w.Opts)); metricsServer != nil {
+			infoLog.Log(logging.MessageKey(), "starting server", "name", w.Metrics.Name, "address", w.Metrics.Address)
+			ListenAndServe(logger, &w.Metrics, metricsServer)
+		}
+
 		return nil
 	})
 }
+
+//getMetricsHandler returns the handler for metrics given a gatherer and prometheus handler options. Zero value
+// arguments are ok in which case prometheus-defined defaults are used.
+func getMetricsHandler(gatherer stdprometheus.Gatherer, opts *promhttp.HandlerOpts) http.Handler {
+	if gatherer == nil {
+		gatherer = stdprometheus.DefaultGatherer
+	}
+
+	if opts == nil {
+		opts = &promhttp.HandlerOpts{}
+	}
+
+	//todo: need to add security layer decoration
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.HandlerFor(gatherer, *opts))
+	return mux
+}
+
+//todo maybe this webpa server arm for metrics does not belong here
```

Comment:
This doesn't belong here.  We'll make a separate metrics package within `webpa-common` for facilitation code.